### PR TITLE
[FIX] web: fix error on creating new record with ref field

### PR DESCRIPTION
**Pull request by yelizariev**
STEPS:
* Open ``Settings >> External Identifiers``
* Click `[Create]`

Patch in `getValue` fixes js error:

> Caused by: TypeError: Cannot read properties of undefined (reading 'model')

Patch in `onWillUpdateProps` fixes infinite loop

task-3047190
opw-3056362

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
Also, opw-3073516

**Review by mcm-odoo — COMMENTED** [review]
It seems ok, can you add a test?

File: addons/web/static/src/views/fields/reference/reference_field.js [hunk 1]
```diff
@@ -31,10 +32,14 @@ export class ReferenceField extends Component {
     }
     getValue(p) {
         if (p.type === "char") {
+            const pdata = this.getPreloadedData(p);
```
[the remark this answers is not in the document]
Comment (yelizariev) [reply addons/web/static/src/views/fields/reference/reference_field.js:35]: We avoid such kind of refactoring in stable. Custom modules may still depend on it.

**Comment by yelizariev** [comment]
The test is added.


Error without any changes:

image

(Infinite loop error is not captured)

**Comment by mcm-odoo** [comment]
@robodoo r+

**Pull request closed without merging**